sdk/digitaltwins/azure-digitaltwins/azure/digitaltwins/_digitaltwins_client.py

```diff
@@ -0,0 +1,926 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import uuid
+from datetime import datetime
+from msrest import Serializer, Deserializer
+from azure.core.paging import ItemPaged
+from azure.core.tracing.decorator import distributed_trace
+from azure.core import MatchConditions
+
+from azure.core.exceptions import (
+    HttpResponseError,
+    ServiceRequestError,
+    ResourceExistsError,
+    ResourceNotFoundError,
+    ResourceModifiedError,
+    ResourceNotModifiedError
+)
+
+from ._utils import (
+    prep_if_match,
+)
+
+from ._generated import models
+from ._generated import AzureDigitalTwinsAPI
+
+class DigitalTwinsClient(object):
+    """Creates an instance of AzureDigitalTwinsAPI.
+
+    :param str endpoint: The URL endpoint of an Azure search service
+    :param ~azure.core.credentials.AzureKeyCredential credential:
+        A credential to authenticate requests to the service
+    """
+    def __init__(self, endpoint, credential, **kwargs: any):
+        # type: (str, AzureKeyCredential, dict) -> None
+
+        client_models = {k: v for k, v in models.__dict__.items() if isinstance(v, type)}
+        self._serialize = Serializer(client_models)
+        self._deserialize = Deserializer(client_models)
+        self.endpoint = endpoint #type: str
+        self.credential = credential #type AzureKeyCredential
+        self._client = AzureDigitalTwinsAPI(
+            credential=credential,
+            base_url=endpoint,
+            **kwargs
+        ) #type: AzureDigitalTwinsAPI
+
+    @distributed_trace
+    def get_digital_twin(self, digital_twin_id, **kwargs):
+        # type: (str, dict) -> object
+        """Get a digital twin.
+
+        :param str digital_twin_id: The Id of the digital twin.
+        :returns: The twin object.
+        :rtype: object
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        :raises :class: `~azure.core.exceptions.ResourceNotFoundError`:
+            If the digital twin doesn't exist.
+        """
+        error_map = {
+            404: ResourceNotFoundError
+        }
+
+        try:
+            return self._client.digital_twins.get_by_id(
+                digital_twin_id,
+                error_map=error_map,
+                **kwargs
+            )
+        except ResourceNotFoundError:
+            return None
```

Comment:
We usually raise a ResoucreNotFoundError - is there a particular scenario you feel returning `None` here would be better?

Reply:
Removed error logic from the convenience layer.